rpcserver.go

```diff
@@ -5067,3 +5075,108 @@ func (r *rpcServer) SubscribeChannelBackups(req *lnrpc.ChannelBackupSubscription
 		}
 	}
 }
+
+// ChannelAcceptor dispatches a bi-directional streaming RPC in which
+// OpenChannel requests are sent to the client and the client responds with
+// a boolean that tells LND whether or not to accept the channel. This allows
+// node operators to specify their own criteria for accepting inbound channels
+// through a single persistent connection. There may only be one client at a
+// time for this RPC.
+func (r *rpcServer) ChannelAcceptor(stream lnrpc.Lightning_ChannelAcceptorServer) error {
+
+	var (
+		acceptorClient *chanacceptor.Client
+		rpcAcceptor    *chanacceptor.RPCAcceptor
+		ok             bool
+	)
+
+	// Locate the RPCAcceptor within the ChainedAcceptor and register this RPC client.
+	openChanPredicate := r.server.fundingMgr.cfg.OpenChannelPredicate
+	chainedAcceptor, _ := openChanPredicate.(*chanacceptor.ChainedAcceptor)
+
+	for _, acceptor := range chainedAcceptor.Acceptors {
+		rpcAcceptor, ok = acceptor.(*chanacceptor.RPCAcceptor)
```

Comment:
To more easily support multiple RPC clients, and also simplify the registration code a little bit, you could create a `NewRPCAcceptor` method that can be called every time this RPC is called. This acceptor would be strictly tied to this rpc context (so no registration/unregistration would be necessary), and would be added to the current chain of acceptors. When the RPC context is done, it would be removed from the chain.

Reply:
Yeah I agree with this approach, I started implementing the other way (multiple clients per RPCAcceptor) and found it more difficult to reason about.